lib/board.rb

```diff
@@ -28,47 +28,53 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinate_array)
+    return false if lengths_differ(ship, coordinate_array)
+    return false if any_coordinates_invalid(coordinate_array)
+    return false if any_coordinates_already_hold_ship(coordinate_array)
+    return false if coordinates_not_consecutive(ship, coordinate_array)
+    # All validations passed, therefore it is a valid placement
+    return true
+  end
 
-    # Test if correct Length
-    return false if ship.length != coordinate_array.length
+  def lengths_differ(ship, coordinate_array)
+    ship.length != coordinate_array.length
+  end
 
-    # Test if Coordinates all valid
-    if coordinate_array.any?{ |coordinate| !valid_coordinate?(coordinate)}
-      return false
-    end
+  def any_coordinates_invalid(coordinate_array)
+    coordinate_array.any?{ |coordinate| !valid_coordinate?(coordinate)}
+  end
 
-    # Test if Ship in any coordinates
-    if coordinate_array.any?{ |coordinate| @cells[coordinate].ship }
-      return false
-    end
+  def any_coordinates_already_hold_ship(coordinate_array)
+    coordinate_array.any?{ |coordinate| @cells[coordinate].ship }
+  end
 
+  def coordinates_not_consecutive(ship, coordinate_array)
     # Test if all in one row or one column
     rows_from_input = coordinate_array.map { |coordinate| coordinate[0]}
     columns_from_input = coordinate_array.map { |coordinate| coordinate[1..-1].to_i}
 
     in_row = rows_from_input.uniq.length == 1
     in_column = columns_from_input.uniq.length == 1
 
-    return false if !in_row && !in_column
+    return true if !in_row && !in_column
```

Comment:
Not sure this logic here should be changed.

Have you run this against the tests?